node/build.rs

```diff
@@ -1,13 +1,13 @@
 #[cfg(not(feature = "download"))]
 mod download {}
 
-#[cfg(any(not(feature = "download"), feature = "doc"))]
+#[cfg(all(not(feature = "download"), feature = "doc"))]
```

Comment:
In 7bd1486e28a9d5d47b031e532001527951a28915:

Now there is no `main` function if neither `doc` nor `download` is provided.

Reply:
I think my local CI would've found this -- I didn't check because the code was visibly wrong -- but it would be nice if Github could too.

In my local CI I have a complicated setup in which I test with no features, all features, every feature individually, and then two pairs of random features (generated from a hash of cargo.toml or something like that). And then after generating all those, I post-filter them and exclude any feature lists for which no feature matches `^[0-9_]*$`.

Nix is a pretty awful language but bash is even worse, so you might not want to try to reproduce that..

Reply:
Thanks man.

Reply:
I added two lines to the `extra_test.sh` script

```rust
    $cargo check --all-features
    $cargo check --doc
```

And a bunch more feature gate ugly hacks to get `compile_error!` to be the only output when using `--no-default-features`.